owncloudApp/src/main/java/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -200,17 +216,86 @@ protected void onCreate(Bundle savedInstanceState) {
+
+                    if (haveMultiAccount) {
+                        if (spacePersonal == null) {
+                            showListFiles();
+                            showRetainerFragment();
+                            updateDirectoryList();
+                            if (mParents.size() == 1) {
+                                updateToolbar(getString(R.string.uploader_top_message));
+                            }
+
+                        } else {
+                            mListView = findViewById(android.R.id.list);
+                            fragmentContainer = findViewById(R.id.fragment_container);
+                            mListView.setVisibility(View.GONE);
+                            fragmentContainer.setVisibility(View.VISIBLE);
+                            initAndShowListOfSpaces();
+                            updateToolbar(getString(R.string.choose_upload_space));
+                            getSupportFragmentManager().setFragmentResultListener(SpacesListFragment.REQUEST_KEY_CLICK_SPACE, this, (requestKey,
+                                                                                                                                     bundle) -> {
+                                OCFile rootSpaceFolder = bundle.getParcelable(SpacesListFragment.BUNDLE_KEY_CLICK_SPACE);
+                                mFile = rootSpaceFolder;
+                                currentSpaceId = mFile.getSpaceId();
+                                showListFiles();
+                                updateDirectoryList();
+                                showRetainerFragment();
+                            });
+                        }
+                    } else { //Sin multicuenta
+                        if (spacePersonal != null) {
+                            initAndShowListOfSpaces();
+                            updateToolbar(getString(R.string.choose_upload_space));
+                            getSupportFragmentManager().setFragmentResultListener(SpacesListFragment.REQUEST_KEY_CLICK_SPACE, this, (requestKey,
+                                                                                                                                     bundle) -> {
+                                OCFile rootSpaceFolder = bundle.getParcelable(SpacesListFragment.BUNDLE_KEY_CLICK_SPACE);
+                                mFile = rootSpaceFolder;
+                                currentSpaceId = mFile.getSpaceId();
+                                showListFiles();
+                                updateDirectoryList();
+                                showRetainerFragment();
+                            });
+                        } else {
+                            showListFiles();
+                            showRetainerFragment();
+                            updateDirectoryList();
+                            updateToolbar(getString(R.string.uploader_top_message));
+                        }
+                    }
```

Comment:
Let's try refactorizing this and first check if we are in a oC10 or oCIS account (`spacePersonal == null`) and then if we have multi-account or not, I think we could save much repeated code 👍

Reply:
Done